File: bench/elixir/collatz.exs
defmodule B do
  def steps(1, s), do: s
  def steps(n, s) when rem(n, 2) == 0, do: steps(div(n, 2), s + 1)
  def steps(n, s), do: steps(3 * n + 1, s + 1)
end

n = String.to_integer(System.get_env("BENCH_N") || "250000")
IO.puts(Enum.reduce(1..(n - 1), 0, fn start, b -> max(b, B.steps(start, 0)) end))

File: bench/erlang/collatz.erl
#!/usr/bin/env escript
%%! -mode(compile)
main(_) ->
    N = bench_n(250000),
    io:format("~w~n", [lists:foldl(fun(Start, B) -> max(B, steps(Start, 0)) end, 0, lists:seq(1, N-1))]).
bench_n(D) -> case os:getenv("BENCH_N") of false -> D; V -> list_to_integer(V) end.
steps(1, S) -> S;
steps(N, S) when N rem 2 =:= 0 -> steps(N div 2, S+1);
steps(N, S) -> steps(3*N+1, S+1).
